--- contrib/check_backup.pl	2002-04-07 07:13:57.000000000 +0200
+++ contrib/check_backup.pl.oden	2008-01-11 15:50:18.000000000 +0100
@@ -24,7 +24,8 @@
 use Getopt::Long;
 use vars qw($opt_H $opt_d $opt_s $opt_t $verbose $PROGNAME);
 use lib $main::runtimedir;
-use utils qw($TIMEOUT %ERRORS &print_revision &usage &support &is_error);
+use lib '/usr/libexec/nagios/plugins';
+use utils qw($TIMEOUT %ERRORS &print_revision &usage &support);
 
 sub help ();
 sub print_help ();
