--- contrib/check_appletalk.pl	2007-08-28 05:19:45.000000000 +0200
+++ contrib/check_appletalk.pl.oden	2008-01-11 15:38:43.000000000 +0100
@@ -22,6 +22,7 @@
 }
 
 use strict;
+use lib $main::runtimedir;
 use lib "/usr/local/nagios/libexec";
 
 use utils qw($TIMEOUT %ERRORS &print_revision &support);
@@ -196,15 +197,3 @@
 }
 print $answer;
 exit $ERRORS{$state};
-
-
-
-
--------------------------------------------------------
-This sf.net email is sponsored by:ThinkGeek
-Welcome to geek heaven.
-http://thinkgeek.com/sf
-_______________________________________________
-Nagios-devel mailing list
-Nagios-devel@lists.sourceforge.net
-https://lists.sourceforge.net/lists/listinfo/nagios-devel
