diff -Naurp nagios-plugins-1.4.11/contrib/check_javaproc.pl nagios-plugins-1.4.11.oden/contrib/check_javaproc.pl
--- nagios-plugins-1.4.11/contrib/check_javaproc.pl	2002-11-18 22:58:58.000000000 +0100
+++ nagios-plugins-1.4.11.oden/contrib/check_javaproc.pl	2008-01-11 18:30:39.000000000 +0100
@@ -28,7 +28,7 @@ use strict;
 use Getopt::Long;
 use vars qw($opt_w $opt_c $verbose $classname);
 use vars qw($PROGNAME);
-use lib "utils.pm" ;
+use lib utils.pm;
 use utils qw($TIMEOUT %ERRORS &print_revision &support &usage);
 
 $PROGNAME="check_javaprocs";
diff -Naurp nagios-plugins-1.4.11/contrib/check_rrd_data.pl nagios-plugins-1.4.11.oden/contrib/check_rrd_data.pl
--- nagios-plugins-1.4.11/contrib/check_rrd_data.pl	2002-02-28 07:42:51.000000000 +0100
+++ nagios-plugins-1.4.11.oden/contrib/check_rrd_data.pl	2008-01-11 18:31:00.000000000 +0100
@@ -41,7 +41,7 @@
         # require "$runtimedir/utils.pm";
 # }
 
-require '/usr/libexec/nagios/plugins/utils.pm';
+require utils.pm;
 use RRD::File;
 # use strict;			# RRD:File and utils.pm don't like this
 
diff -Naurp nagios-plugins-1.4.11/contrib/check_snmp_printer.pl nagios-plugins-1.4.11.oden/contrib/check_snmp_printer.pl
--- nagios-plugins-1.4.11/contrib/check_snmp_printer.pl	2004-01-18 21:07:01.000000000 +0100
+++ nagios-plugins-1.4.11.oden/contrib/check_snmp_printer.pl	2008-01-11 18:30:45.000000000 +0100
@@ -82,7 +82,7 @@ use strict;
 use Getopt::Long;
 use vars qw($opt_V $opt_h $opt_H $opt_P $opt_t $opt_d $session $error $answer $key
    $response $PROGNAME $port $hostname );
-use lib  "utils.pm";
+use lib utils.pm;
 use utils qw(%ERRORS &print_revision &support &usage );
 use Net::SNMP;
 
